dwave/cloud/auth/flows.py

```diff
@@ -202,3 +222,72 @@ def from_config_model(cls, config: ClientConfig, **kwargs) -> LeapAuthFlow:
             authorization_endpoint=authorization_endpoint,
             token_endpoint=token_endpoint,
             session_config=session_config)
+
+    def run_oob_flow(self):
+        """Run the OAuth 2.0 Authorization Code exchange flow, using the out of
+        band code exchange.
+
+        After authorizing access by visiting the authorization URL displayed,
+        user has to copy the authorization code and paste it back in their
+        terminal. On successful completion Leap API access and refresh tokens
+        are returned.
+        """
+        self.redirect_uri = self._OOB_REDIRECT_URI
+
+        url = self.get_authorization_url()
+        print(self._VISIT_AUTH_MSG, url)
+
+        code = input(self._INPUT_CODE_MSG)
+        return self.fetch_token(code=code)
```

Comment:
This method does not work for me (the `run_redirect_flow` works). I get a URL, I can open the URL, it accepts clicks on both the `authorize` and `decline` buttons, but neither causes the browser to progress to the next page that I expected to return an authorization code.

Reply:
That's expected, as Leap doesn't implement OOB (yet). You can test it if you change `redirect_uri=oob`, in which case you'll be redirected to a non-existing Leap page, but you'll at least be able to copy `code` from the URL query.